src/fmelt.c

```diff
@@ -105,7 +105,7 @@ SEXP measurelist(SEXP measure, SEXP dtnames) {
     SEXP x = VECTOR_ELT(measure, i);
     switch(TYPEOF(x)) {
       case STRSXP  :
-        SET_VECTOR_ELT(ans, i, chmatch(x, dtnames, 0));
+        SET_VECTOR_ELT(ans, i, chmatch(x, dtnames, NA_INTEGER));
```

Comment:
The new `chmatch_na` function in this PR that I just removed I initially thought was necessary because this PR needed `NA`s in `x` not to match to `NA`s in `table`. So I started to add `bool matchNAtoNA` argument to `chmatchMain`. Then it turned out that just passing `nomatch=NA_INTEGER` on this line (the only line where `chmatch_na` was used) instead of `nomatch=0`, seemed to work and passed the new tests. I left `matchNAtoNA` as a comment in `chmatchMain` in case we need to come back to that in future.
@tdhock Please check and let me know if I got anything wrong here. If so, a new test is probably needed please.